src/java/org/apache/cassandra/db/memtable/TrieMemtable.java

```diff
@@ -360,8 +360,8 @@ public FlushablePartitionSet<MemtablePartition> getFlushSet(PartitionPosition fr
     {
         boolean allPositionsToFlush = from == null && to == null
                                       || from != null && to != null
-                                         && from.isMinimum()
-                                         && to.isMaximum();
+                                         && from.isMinimum() && from.kind() == PartitionPosition.Kind.MIN_BOUND
+                                         && to.isMaximum() && to.kind() == PartitionPosition.Kind.MAX_BOUND;
```

Comment:
To clarify (or add to the confusion), the maximum is sometimes out of scope (`RandomPartitioner`) and other times isn't (`Murmur3Partitioner`). The `MAX_BOUND` check is necessary for the maximum, but it should be part of the `isMaximum` call rather than done here.